### PR TITLE
PrimaryVertex sorter fix and store of the score vector - 74X version

### PR DESCRIPTION
**Pull request by arizzi**
This PR fix a bug in the filling of the PV sorter association map and adds the requested feature of storing the score vector (the modified sumpt2)
This is needed by miniaod
(back port of #8101 )

@VinInn @gpetruc

### CODE REVIEW

**Comment by slava77** [comment]
Hi Andrea, 

please clarify in which use case this bug manifests itself; more specifically, would it change the standard reco outputs downstream from this producer; does it change physics performance in miniAOD.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by arizzi** [comment]
The bug is simply that the edm::Association for the "original" collection is not properly initialized (but this is not used in RECO).
In addition the valuemap is stored, this add one new collection (100bytes/event) in RECO and AOD.

Both this features are needed for the stuff we are preparing for miniAOD in 74X.

**Comment by slava77** [comment]
thanks.
I was mainly looking for a more descriptive documentation for the change

File: CommonTools/RecoAlgos/plugins/PrimaryVertexSorter.h [hunk 1]
```diff
@@ -193,10 +201,11 @@ void PrimaryVertexSorter<ParticlesCollection>::produce(Event& iEvent,  const Eve
 
 
   if(produceOriginalMapping_){
-    auto_ptr< CandToVertex>  pfCandToOriginalVertexOutput( new CandToVertex );
+    auto_ptr< CandToVertex>  pfCandToOriginalVertexOutput( new CandToVertex(vertices) );
```
Comment (arizzi) [inline CommonTools/RecoAlgos/plugins/PrimaryVertexSorter.h]: @slava77 this is the bug

File: CommonTools/RecoAlgos/plugins/PrimaryVertexSorter.h [hunk 2]
```diff
@@ -98,11 +99,13 @@ PrimaryVertexSorter<ParticlesCollection>::PrimaryVertexSorter(const edm::Paramet
   if(produceOriginalMapping_){
       produces< CandToVertex> ("original");
       produces< CandToVertexQuality> ("original");
+      produces< VertexScore> ("original");
   }
   if(produceSortedVertices_){
       produces< reco::VertexCollection> ();
       produces< CandToVertex> ();
       produces< CandToVertexQuality> ();
+      produces< VertexScore> ();
```
Comment (arizzi) [inline CommonTools/RecoAlgos/plugins/PrimaryVertexSorter.h]: @slava77 this is the additional product in RECO/AOD with current RECO/AOD config

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/8102	729108a
tested in CMSSW_7_4_0_pre8 /test area sign526/

currently monitored quantities are unchanged, based on extended short matrix tests (~100 events in ~20 different workflows, data and MC; looking at RECO outputs in DQM and FWlite)

two new products show up
-       floatedmValueMap_offlinePrimaryVerticesWithBS__reRECO.
-       floatedmValueMap_offlinePrimaryVertices__reRECO.

Entries in the maps make sense: here's a scan of one event with the score values and nTracks for the PVs in the event

```
Scan("floatedmValueMap_offlinePrimaryVertices__reRECO.obj.values_:recoVertexs_offlinePrimaryVertices__reRECO.obj.nTracks()")
***********************************************
*    Row   * Instance * floatedmV * recoVerte *
***********************************************
*        0 *        0 * 518.17297 *        73 *
*        0 *        1 * 22.402217 *        22 *
*        0 *        2 * 17.319988 *        27 *
*        0 *        3 * 12.569139 *        14 *
*        0 *        4 * 12.214971 *        15 *
*        0 *        5 * 7.3361372 *         7 *
*        0 *        6 * 6.3250789 *         7 *
*        0 *        7 * 6.1019206 *        10 *
*        0 *        8 * 5.5338187 *        15 *
*        0 *        9 * 5.1360631 *         7 *
*        0 *       10 * 3.6782689 *         5 *
*        0 *       11 * 2.9102571 *         6 *
*        0 *       12 * 2.6730599 *         3 *
*        0 *       13 * 2.1457858 *        11 *
*        0 *       14 * 1.0911568 *         4 *

```

**Comment by davidlange6** [comment]
+1

**Pull request merged**